CONTRIBUTING.md

```diff
@@ -1,35 +1,152 @@
-# How to contribute
+# Contributing to TiKV
 
-This document outlines some of the conventions on development workflow, commit message formatting, contact points and other
-resources to make it easier to get your contribution accepted.
+Thanks for your interest in contributing to TiKV! This document outlines some of the conventions on building, running, and testing TiKV, the development workflow, commit message formatting, contact points and other resources.
 
-## Getting started
+If you need any help or mentoring getting started, understanding the codebase, or making a PR (or anything else really), please ask on [Slack](https://join.slack.com/t/tikv-wg/shared_invite/enQtNTUyODE4ODU2MzI0LTgzZDQ3NzZlNDkzMGIyYjU1MTA0NzIwMjFjODFiZjA0YjFmYmQyOTZiNzNkNzg1N2U1MDdlZTIxNTU5NWNhNjk), [Discourse](https://forum.tikv.org/c/tikv), or [WeChat](./README.md#WeChat).
+
+## Building and setting up a development workspace
+
+TiKV is mostly written in Rust, but has components written in C++ (RocksDB) and Go (gRPC). We are currently using the Rust nightly toolchain. To guarantee consistency in coding style, we use linters and automated formatting tools. 
+
+### Prerequisites
+
+To build TiKV you'll need to at least have the following installed:
+
+* `git` - Version control
+* [`rustup`](https://rustup.rs/) - Rust installer and toolchain manager
+* `make` - Build tool (run common workflows)
+* `cmake` - Build tool (required for gRPC)
+* `awk` - Pattern scanning/processing language
+
+If you are targeting platforms other than x86_64 linux, you'll also need:
+
+* [`llvm` and `clang`](http://releases.llvm.org/download.html) - Used to generate bindings for different platforms and build native libraries (required for grpcio, rocksdb).
+
+### Getting the repository
+
+```
+git clone https://github.com/tikv/tikv.git
+cd tikv
+# Future instructions assume you are in this directory
+```
+
+### Configuring your Rust toolchain
+
+`rustup` is the official toolchain manager for Rust, similar to `rvm` or `rbenv` from the Ruby world.
+
+TiKV is pinned to a version of Rust using a `rust-toolchain` file. `rustup` and `cargo` will automatically use this file. We also use the `rustfmt` and `clippy` components, to install those:
+
+```bash
+rustup component add rustfmt
+rustup component add clippy
+```
+
+### Building and testing
+
+TiKV includes a `Makefile` with common workflows, you can also use `cargo`, as you would in many other Rust projects.
+
+At this point, you can build TiKV:
+
+```bash
+make build
+```
+
+During interactive development, you may prefer using `cargo check`, which will parse, borrow check, and lint your code, but not actually compile it:
+
+```bash
+cargo check --all
```

Comment:
Does this still cover everything because of #5289 ?

Reply:
Yes, the issue there is just that the binaries are not in the main crate, they are still in a crate which is part of the workspace